server/server.go

```diff
@@ -955,3 +947,37 @@ func (*Server) configureBackends(backends map[string]*types.Backend) {
 		}
 	}
 }
+
+// newMetrics instantiates the proper Metrics implementation, depending on the global configuration.
+// Note that given there is no metrics instrumentation configured, it will return nil.
+func newMetrics(globalConfig GlobalConfiguration, name string) middlewares.Metrics {
+	metricsEnabled := globalConfig.Web != nil && globalConfig.Web.Metrics != nil
+	if metricsEnabled && globalConfig.Web.Metrics.Prometheus != nil {
+		metrics, _, err := middlewares.NewPrometheus(name, globalConfig.Web.Metrics.Prometheus)
+		if err != nil {
+			log.Errorf("Error creating Prometheus Metrics implementation: %s", err)
+			return nil
+		}
+		return metrics
+	}
+
+	return nil
+}
+
+func registerRetryMiddleware(
```

Comment:
I basically introduced to enable testing, that the middleware is constructed properly. I think this would not be possible anymore, when I inline the logic back into the `loadConfig`. So I can remove it of course if you prefer it this way, but I guess then the test also has to die.